src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java

```diff
@@ -58,6 +61,200 @@ public static void exportAllHistoricalToFile(
+    exporter.close();
+
+    // Check for Circularized Event
+    if (EventFile.DidFileCircularizedEventOccur()) {
+      throw new CircularizedFileException("A circularized event was found in the event logs.");
+    }
+    return dataPoints;
+  }
+
+  /**
+   * Executes EBD call, waits for data return.
+   *
+   * @param ebdStr EBD string
+   * @return EBD Exporter - caller must close exporter
+   * @throws TimeoutException when there is no response before timeout period
+   * @throws IOException when there is an Exporter Exception
+   */
+  public static Exporter ExecuteEBDCall(String ebdStr) throws IOException, TimeoutException {
+    final long start = System.currentTimeMillis();
+    final Exporter exporter = new Exporter(ebdStr);
+    int available = exporter.available();
+    // check on data availability
+    while (available == 0) {
+      // sleep
+      try {
+        Thread.sleep(HistoricalDataConstants.THREAD_SLEEP_MS);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+      // check for timeout
+      if (System.currentTimeMillis() - start > HistoricalDataConstants.MAX_EBD_WAIT) {
+        exporter.close();
+        throw new TimeoutException("could not wait any longer for EBD call");
+      }
+
+      // check for data available
+      available = exporter.available();
+    }
+    return exporter;
+  }
+
+  /**
+   * Reads historical log between <code>startTime</code> and <code>endTime</code>. Returns
+   * Datapoints.
+   *
+   * @param startTime start time of export
+   * @param endTime end time of export
+   * @param includeTagGroupA include tag group A
+   * @param includeTagGroupB include tag group B
+   * @param includeTagGroupC include tag group C
+   * @param includeTagGroupD include tag group D
+   * @param stringHistorical export string historical logs if true
+   * @return data points from response
+   * @throws IOException if export block descriptor fails
+   * @throws JSONException if unable to parse int to string enumeration file
+   * @throws CircularizedFileException if circularized file event is found
+   */
+  public static ArrayList readHistoricalFifo(
+      String startTime,
+      String endTime,
+      boolean includeTagGroupA,
+      boolean includeTagGroupB,
+      boolean includeTagGroupC,
+      boolean includeTagGroupD,
+      boolean stringHistorical)
+      throws IOException, JSONException, CircularizedFileException {
+
+    // create EBD string
+    final String ebdStr =
+        prepareHistoricalFifoReadEBDString(
+            startTime,
+            endTime,
+            includeTagGroupA,
+            includeTagGroupB,
+            includeTagGroupC,
+            includeTagGroupD,
+            stringHistorical);
+
+    // Execute EBD call and parse results
+    try {
+      final Exporter exporter = ExecuteEBDCall(ebdStr);
+      return parseEBDHistoricalLogExportResponse(exporter);
+    } catch (TimeoutException e) {
+      return new ArrayList();
```

Comment:
No logging of the timeout?

Reply:
I don't think the proper way to handle a timeout is in this library. It prevents the connector from having visibility into the timeout and instead it looks like it just gets a list of no datapoints. Shouldn't the connector that implements it be able to catch the TimeoutException and either log it or retry?

Reply:
Have you tested this on the Flexy? TimeoutException reports as Java 1.5 or newer: 
![image](https://user-images.githubusercontent.com/51409174/152416295-f2de8162-8788-4443-868c-affd03734efe.png)

Reply:
👍🏼  :(  you are correct. Why can't we have nice things?

Reply:
Changed to custom timeout class.